lib/linter-swiftlint.coffee

```diff
@@ -1,40 +1,54 @@
-linterPath = atom.packages.getLoadedPackage("linter").path
-Linter = require "#{linterPath}/lib/linter"
-path = require 'path'
+module.exports = LinterSwiftlint =
+  activate: ->
+    console.log 'activate linter-swiftlint'# if atom.inDevMode()
+    unless atom.packages.getLoadedPackages 'linter-plus'
+      @showError '[Linter+ swiftlint] `linter-plus` package not found,
+       please install it'
 
-module.exports = class LinterSwiftlint extends Linter
-  # The syntax that the linter handles. May be a string or
-  # list/tuple of strings. Names should be all lowercase.
-  @syntax: ['source.swift']
-  # A string, list, tuple or callable that returns a string, list or tuple,
-  # containing the command line (with arguments) used to lint.
-  cmd: 'swiftlint lint'
-  linterName: 'swiftlint'
-  errorStream: 'stdout'
-  # A regex pattern used to extract information from the executable's output.
-  # Beacuse swiftc is essentially a specialized clang, I am using the regex from
-  #   from linter-clang.
-  regex: '.+:(?<line>\\d+): ((?<error>error)|(?<warning>warning)): ' +
-    '(?<message>.*)'
+  showError: (message = '') ->
+    atom.notifications.addError message
 
-  constructor: (editor) ->
-    super(editor)
-    @listen = atom.config.observe 'linter-swiftlint.liveLinting', (value) =>
-      @lintLive = value
+  provideLinter: ->
+    {
+      scopes: ['source.swift']
+      lint: @lint
+      lintOnFly: true
```

Comment:
shouldn't this be false? 'cause you are giving the file to the program and `lintOnFly` means that your validator accepts the current text buffer as stdin and lints it. Which doens't seem to be the case here.